src/Sulu/Bundle/PageBundle/Tests/Unit/EventListener/DomainEventSubscriberTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\PageBundle\Tests\Unit\EventListener;
+
+use PHPUnit\Framework\TestCase;
+use Sulu\Bundle\ActivityBundle\Application\Collector\DomainEventCollector;
+use Sulu\Bundle\ActivityBundle\Application\Dispatcher\DomainEventDispatcher;
+use Sulu\Bundle\DocumentManagerBundle\Collector\DocumentDomainEventCollector;
+use Sulu\Bundle\PageBundle\Document\PageDocument;
+use Sulu\Bundle\PageBundle\Domain\Event\PageRemovedEvent;
+use Sulu\Bundle\PageBundle\EventListener\DomainEventSubscriber;
+use Sulu\Component\DocumentManager\DocumentManager;
+use Sulu\Component\DocumentManager\Event\RemoveEvent;
+use Sulu\Component\DocumentManager\NamespaceRegistry;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+
+class DomainEventSubscriberTest extends TestCase
+{
+    public function testAddResourceSegmentToPageRemovedEvent(): void
+    {
+        $pageDocument = new PageDocument();
+        $pageDocument->setTitle('Cool page');
+        $pageDocument->setLocale('en_US');
+        $pageDocument->setResourceSegment('/cool-page');
+        (function () {
+            $this->uuid = '6860560d-5f90-445b-b263-7e88ce03db4d';
+            $this->webspaceName = 'My Webspace';
+        })->bindTo($pageDocument, get_class($pageDocument))();
```

Comment:
We did not yet use `bindTo` in any tests. Would just call the setters or why is the bindTo required?

Reply:
As you can see I didn't use mocking in this test and persistent felt like an overkill, so I used a closure, which is an alternative for using reflection api for such cases.

Reply:
> And in other places the documents are persisted, so the values are set by Doctrine using reflection.

We did more and more avoid mocks lately there are sadly still to many tests using mocks where they are not even required. So we did use in newer code mostly the Trait when there was no constructor or setter method: https://github.com/sulu/sulu/blob/2.5/src/Sulu/Bundle/TestBundle/Testing/SetGetPrivatePropertyTrait.php which make to code so more decoupled from mocking libraries. We still use mocks to mock things like Repository calls away. But where possible we try to avoid them. Personally I'm real fan of https://blog.frankdejonge.nl/testing-without-mocking-frameworks/ but we are definitely not there yet with the codebase.

Reply:
I also prefer to avoid mocking, at least the excessive use.
That trait is exactly what's needed here.

Reply:
Replaced the closure with the trait methods.